.github/workflows/test.yml

```diff
@@ -123,6 +123,24 @@ jobs:
           target-ros2-distro: foxy
           vcs-repo-file-url: "${{ github.workspace }}/.github/workflows/repo_file_for_test_cpp_with_dependency.repos"
 
+  test_private_repo:
+    name: "Test with private repos"
+    runs-on: ubuntu-20.04
+    needs: pre_condition
+    # We don't have access to the secret token for the test repo on a fork
+    if: ${{ !github.event.repository.fork && !github.event.pull_request.head.repo.fork }}
+    steps:
+      - uses: actions/checkout@v2
+      - uses: ros-tooling/setup-ros@0.1.2
+        with:
+          required-ros-distributions: foxy
+      - uses: ./
+        with:
+          import-token: ${{ secrets.TOKEN_PRIVATE_REPO_TEST }}
```

Comment:
i think you may be able to just use GITHUB_TOKEN here?

Reply:
I'll try but from what I can tell it doesn't work.

Reply:
yeah I don't think it works. From what I can tell `GITHUB_TOKEN` is for the current repo only.

Reply:
@emersonknapp @christophebedard Hello, thank you for working on this feature!

I've forked and tested this in https://github.com/kenji-miyake/action-ros-ci/pull/2.
As @christophebedard says, `GITHUB_TOKEN` is for the current repo and couldn't be overridden.
So I generated a personal token from [here](https://github.com/settings/tokens/new) and added it as `secrets.ACTION_ROS_CI_TOKEN` from [here](https://github.com/kenji-miyake/action-ros-ci/settings/secrets/actions).

![image](https://user-images.githubusercontent.com/31987104/117562187-fc29ff00-b0d7-11eb-95f8-6b25b1b37bbc.png)

After that, CI seems to succeed.
![image](https://user-images.githubusercontent.com/31987104/117562142-b2411900-b0d7-11eb-87b0-1b186be2b020.png)

In ros-tooling org, I believe a personal token can be replaced with [organization secrets](https://github.blog/changelog/2020-05-14-organization-secrets/) or something else.

Reply:
@kenji-miyake thanks for confirming!

Since this PR actually contains both a test and a fix for private repos, we should get this merged!

@emersonknapp could you look into creating an organization secret and adding it to this repo as `TOKEN_PRIVATE_REPO_TEST`? I can't do it (since I'm not an owner I think).

Reply:
Organization Secrets are just a way to set a Secret at the level of the Organization, which individual repositories can then use. The contents of the secret itself are still a personal access token. I don't see a need for this particular token to be used outside this test - so I've added it just to this repo. It has "repo" permissions only - so it should be able to check out private repositories.

I updated the branch which is restarting the workflows - we'll see what happens when the queue clears out. Thanks for bumping that dependabot config down to weekly - what a hassle this backed up queue is. I'm still investigating more ways to thin it out even further.

Reply:
the test passed! :confetti_ball:

but now the branch is out of date :laughing: is there any way to override that setting and squash+rebase anyway?